src/ploneintranet/workspace/browser/tiles/templates/sidebar.pt

```diff
@@ -258,35 +257,94 @@ Supported types include (class names):
+              <a title="Create new task"
+                 class="create-document pat-modal icon-doc-text" data-pat-modal="class: large" data-pat-inject="source:#content"
+                 tal:attributes="href string:${ws/absolute_url}/add_task"
+                 tal:condition="view/can_add">Create task</a>
+            </div>
+          </div>
+          <tal:no-tasks condition="python:not ws.is_case and not tasks ">
             <p i18n:translate="no-task-found">No tasks created yet</p>
           </tal:no-tasks>
-          <tal:tasks condition="tasks">
-            <form action="#"
+          <tal:tasks condition="python:ws.is_case or tasks">
+            <form method="post" class="pat-autosubmit pat-inject"
                   tal:define="ws view/workspace"
-                  tal:attributes="action string:${ws/absolute_url}/@@sidebar.default"
-                  method="post" class="pat-autosubmit pat-inject" data-pat-inject="target: #workspace-settings > .tabs-content; source: #workspace-settings > .tabs-content && target: #document-content::before; source: .sidebar-status-message">
-              <fieldset class="object-list tasks pat-checklist">
-                <tal:task repeat="task tasks">
-                  <br/>
-                  <label>
-                    <input type="hidden" name="current-tasks:list" tal:attributes="value python: task['id']"/>
-                    <input type="checkbox" name="active-tasks:list"
-                           tal:attributes="value python: task['id']"
-                           tal:condition="python: task['checked']" checked="checked"/>
-                    <input type="checkbox" name="active-tasks:list"
-                           tal:attributes="value python: task['id']"
-                           tal:condition="python: not task['checked']"/>
-                    <a href="#"
-                       tal:attributes="title python: task['title'];
-                                       href python: task['url']"
-                       tal:content="python:task['content']"></a>
+                  tal:attributes="action string:${ws/absolute_url}/@@sidebar.default#global-statusmessage">
+
+              <tal:workspace_tasks tal:condition="not: ws/is_case">
+                <fieldset class="object-list tasks pat-checklist">
+                  <tal:task repeat="task tasks">
+                    <br/>
+                    <label>
+                      <input type="hidden" name="current-tasks:list" tal:attributes="value task/id"/>
+                      <input type="checkbox" name="active-tasks:list"
+                             tal:attributes="value task/id;
+                                             checked python: 'checked' if task['checked'] else ''"
+                             />
+                      <a class="pat-inject"
+                         tal:attributes="title task/title;
+                                         href python: task['url']+'#document-body'"
```

Comment:
Ideally don't link directly to anchors with pat-inject, because it makes the URLs ugly and doesn't make sense for non-JS use. Instead use the data-pat-inject attribute to select the source/target ids.